### PR TITLE
Ability to rename and change todo status was added

### PR DESCRIPTION
**Pull request by Yevheniya-Deryka**
[DEMO LINK](https://Yevheniya-Deryka.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by grafit-off — APPROVED** [review]
All tasks were completed, and no bugs were recognized, so LGTM ⭐

File: src/App.tsx [hunk 1]
```diff
@@ -1,16 +1,142 @@
...
+  const handleChangeTodoStatus = useCallback(async (todo: Todo) => {
+    const { id, completed } = todo;
+
+    handleSetError({ status: false });
+
+    try {
+      setChangingIds(currValue => [...currValue, id]);
+
+      await changeTodo(id, { completed: !completed });
+      await handleLoadTodos();
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:115]: same here, you can update this data locally, no need to load todos on every change

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,142 @@
...
+    setTimeout(() => {
+      setHasError({ status: false });
+    }, 3000);
+  }, []);
+
+  const handleLoadTodos = useCallback(async () => {
+    try {
+      setHasError({ status: false });
+
+      const todosFromServer = await getTodos((user as User).id);
```
[the remark this answers is not in the document]
Comment (Yevheniya-Deryka) [reply src/App.tsx:43]: Can we use it if we 100% sure that we show this App only if we have user? Or even in this case it's better to add additional 'if' with check for not null user?

[the remark this answers is not in the document]
Comment (OleksiiBlezniuk) [reply src/App.tsx]: Better to add additional if, or don't show page if we don't have user (I mean to move logic from app to child component and pass user as prop there if this user is exist)

As for me first option sounds more simple 🙂

[the remark this answers is not in the document]
Comment (Yevheniya-Deryka) [reply src/App.tsx]: ok! thanks!) done ✔️

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          status={status}
+        />
+
+        <button
+          data-cy="ClearCompletedButton"
+          type="button"
+          className="todoapp__clear-completed"
+          style={(complitedTodos.length === 0)
+            ? { visibility: 'hidden' }
+            : {}}
```
Comment (OleksiiBlezniuk) [inline src/components/TodoList/TodoList.tsx:108]: why not class?

**Review by OleksiiBlezniuk — APPROVED** [review]

File: src/components/TodoList/TodoList.tsx (hunk 3 above)
Comment (Yevheniya-Deryka) [reply src/components/TodoList/TodoList.tsx:108]: can't give you a good reason 😁

**Pull request closed without merging**